datasets/glossis/interpolate_vtk.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import pathlib
+from datetime import datetime
+import pandas as pd
+from tvtk.api import tvtk
+from tvtk.common import configure_input, configure_source_data
+from tvtk.common import configure_connection, configure_input, configure_input_data, configure_outputs
+import gridded 
+import xarray as xr
+import scipy.interpolate
+import numpy as np
+import netCDF4
+import matplotlib.pyplot as plt
+from cmcrameri import cm
+import rasterio
+from rasterio.transform import from_bounds
+
+from utils import download_netcdfs_from_bucket
+
+logger = logging.getLogger(__name__)
+
+def grid2poly(grid):
+    """ Read grid and convert to polydata """
+    points = grid.nodes
+    
+    face_coords_mask = np.dstack([grid.faces.mask, grid.faces.mask])
+    face_coords = np.ma.masked_array(
+        grid.nodes[grid.faces], 
+        face_coords_mask
+    )
+
+    # select faces that we need to duplicate
+    # TODO: duplicate these (with nodes moved to other end of the world) (mod something...)
+    x_max, y_max = face_coords.max(axis=(1)).T
+    x_min, y_min = face_coords.min(axis=(1)).T
+    mask = (x_min < -150) & (x_max > 150) 
+
+    # Select faces
+    faces = grid.faces[~mask]
+    n_cells = faces.shape[0]
+    cell_array = tvtk.CellArray()
+
+    counts = (~faces.mask).sum(axis=1)
+    assert faces.min() >= 0, 'expected 0 based faces'
+    cell_idx = np.c_[counts, faces.filled(-999)].ravel()
+    cell_idx = cell_idx[cell_idx != -999]
+    cell_array.set_cells(n_cells, cell_idx)
+
+    # fill in the properties
+    polydata = tvtk.PolyData()
+    # fill in z dimension
+    polydata.points = np.c_[points, np.zeros_like(points[:, 0])]
+    polydata.polys = cell_array
+        
+    return polydata, mask 
+
+def update_data(path, polydata, mask, variables, t=0):
+    """ Add arrays for each variable to polydata """
+    ds = netCDF4.Dataset(path)
+    for key in variables:
+        arr = ds.variables[key][t, ...]
+
+        array = tvtk.FloatArray()
+        array.from_array(arr[~mask])
+        array.name = key
+        polydata.cell_data.add_array(array)
+    polydata.modified()
+    ds.close()
+
+def setup_pipeline(polydata, width=1000, height=1000):
+    """ VTk pipeline to resample data """
+    resample = tvtk.ResampleToImage()
+    cell2point = tvtk.CellDataToPointData()
+    # setup the pipeline
+    configure_input(cell2point, polydata)
```

Comment:
These are configuring steps? Like creating an edge, a link between these workflow steps?

Reply:
Maybe good to rename these function during import to `vtk_configure_step` or similar. Or only import vtk, so it needs the prefix.

Reply:
This is a so called "vtk Pipeline" this is also in the comment of this function. I could rename this, but everything in this function is vtk related isn't that clear from the comment?